src/wp-admin/includes/class-wp-list-table.php

```diff
@@ -184,27 +185,43 @@ public function __get( $name ) {
 		if ( in_array( $name, $this->compat_fields, true ) ) {
 			return $this->$name;
 		}
+
+		trigger_error(
+			"The property `{$name}` is not defined. Getting a dynamic (undefined) property is " .
+			'deprecated since version 6.4.0! Instead, define the property on the class.',
+			E_USER_DEPRECATED
+		);
+		return null;
 	}
 
 	/**
 	 * Makes private properties settable for backward compatibility.
 	 *
 	 * @since 4.0.0
+	 * @since 6.4.0 Setting a dynamic property is deprecated.
 	 *
 	 * @param string $name  Property to check if set.
 	 * @param mixed  $value Property value.
-	 * @return mixed Newly-set property.
 	 */
 	public function __set( $name, $value ) {
 		if ( in_array( $name, $this->compat_fields, true ) ) {
-			return $this->$name = $value;
+			$this->$name = $value;
+			return;
 		}
+
+		trigger_error(
+			"The property `{$name}` is not defined. Setting a dynamic (undefined) property is " .
+			'deprecated since version 6.4.0! Instead, define the property on the class.',
+			E_USER_DEPRECATED
+		);
+		return null;
```

Comment:
It doesn't seem like the `__set` method should return any values, not even `null`. What's your opinion on that, @hellofromtonya?

```suggestion
```